nano/lib/timer.hpp

```diff
@@ -0,0 +1,250 @@
+#pragma once
+
+namespace nano
+{
+enum timer_state
+{
+	started,
+	stopped
+};
+
+/** Timer utility with nesting support */
+template <typename UNIT = std::chrono::milliseconds>
+class timer
```

Comment:
Would be nice to have unit tests for this class

Reply:
I've added a few tests.